### PR TITLE
Add forms endpoints

### PR DESCRIPTION
**Pull request by rasmusjp**
Added support for the new forms endpoints

### CODE REVIEW

File: src/Umbraco.Headless.Client.Net/Management/FormService.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Refit;
+using Umbraco.Headless.Client.Net.Configuration;
+using Umbraco.Headless.Client.Net.Management.Models;
+
+namespace Umbraco.Headless.Client.Net.Management
+{
+    internal class FormService : IFormService
```
Comment (sitereactor) [inline src/Umbraco.Headless.Client.Net/Management/FormService.cs]: Any particular reason why its called `FormService` and not just `Forms` ?

File: src/Umbraco.Headless.Client.Net/Management/ContentManagementService.cs [hunk 2]
```diff
@@ -80,6 +82,7 @@ public ContentManagementService(IApiKeyBasedConfiguration configuration)
 
         public IContentService Content { get; }
         public IDocumentTypeService DocumentType { get; }
+        public IFormService FormService { get; }
```
Comment (sitereactor) [inline src/Umbraco.Headless.Client.Net/Management/ContentManagementService.cs]: Wouldn't it be more consistent with the rest of the "services" if this was simply called Forms ?

**Comment by sitereactor** [comment]
What happens when validation fails?
Is an exception thrown with a validation error or status code?

File: src/Umbraco.Headless.Client.Net/Management/FormService.cs (hunk 1 above)
Comment (rasmusjp) [reply src/Umbraco.Headless.Client.Net/Management/FormService.cs]: The other classes in management is also called `Service`

File: src/Umbraco.Headless.Client.Net/Management/ContentManagementService.cs (hunk 2 above)
Comment (rasmusjp) [reply src/Umbraco.Headless.Client.Net/Management/ContentManagementService.cs]: Must've been automplete 😅 I've changed it to `Forms`

File: src/Umbraco.Headless.Client.Net/Management/FormService.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            _restService ?? (_restService = RestService.For<FormManagementEndpoints>(_httpClient, _refitSettings));
+
+        public async Task<IEnumerable<Form>> GetAll()
+        {
+            var forms = await Service.GetRoot(_configuration.ProjectAlias);
+            return forms.Forms.Items;
+        }
+
+        public Task<Form> GetById(Guid id) => Service.ById(_configuration.ProjectAlias, id);
+        public Task SubmitEntry(Guid id, object data) => Service.SubmitEntry(_configuration.ProjectAlias, id, data);
```
Comment (sitereactor) [inline src/Umbraco.Headless.Client.Net/Management/FormService.cs]: Wondering if this shouldn't just be "Submit" or "SaveEntry"

File: test/Umbraco.Headless.Client.Net.Tests/Management/FormServiceFixture.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+using Newtonsoft.Json;
+using Refit;
+using RichardSzalay.MockHttp;
+using Umbraco.Headless.Client.Net.Configuration;
+using Umbraco.Headless.Client.Net.Management;
+using Xunit;
+
+namespace Umbraco.Headless.Client.Net.Tests.Management
+{
+    public class FormServiceFixture
```
Comment (sitereactor) [inline test/Umbraco.Headless.Client.Net.Tests/Management/FormServiceFixture.cs]: Missing a few test cases for when submitting an entry fails due to validation or due to some other error.

File: src/Umbraco.Headless.Client.Net/Management/FormService.cs (hunk 3 above)
Comment (rasmusjp) [reply src/Umbraco.Headless.Client.Net/Management/FormService.cs]: If we rename I think we should go for `SaveEntry`

**Comment by rasmusjp** [comment]
> What happens when validation fails?
> Is an exception thrown with a validation error or status code?

I guess Refit throws an exception like it would do for the other endpoints too.

File: test/Umbraco.Headless.Client.Net.Tests/Management/FormServiceFixture.cs (hunk 4 above)
Comment (rasmusjp) [reply test/Umbraco.Headless.Client.Net.Tests/Management/FormServiceFixture.cs]: Added a test for when an error ocours when submitting an entry

File: src/Umbraco.Headless.Client.Net/Management/FormService.cs (hunk 1 above)
Comment (sitereactor) [reply src/Umbraco.Headless.Client.Net/Management/FormService.cs]: yes, sorry, I confused myself. This naming around Forms vs FormService was mostly around the property name in `ContentManagementService`, which you fixed already.

File: src/Umbraco.Headless.Client.Net/Management/FormService.cs (hunk 3 above)
Comment (sitereactor) [reply src/Umbraco.Headless.Client.Net/Management/FormService.cs]: 🤔 lets just keep it as `SubmitEntry`. This is the client library so I guess its more correct that it submits the entry rather then doing the actual saving 😄

**Comment by sitereactor** [comment]
I'll add my stamp of :shipit: here 😄

**Pull request merged**